### PR TITLE
ddl: improve the stability of real tikv test of adding index

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #41452

Problem Summary:

 - Timed out because of the unreasonable huge unit test
 - Dead loop because of the incorrect usage of `context.Background()`

### What is changed and how it works?

Refine the code to fix for the above problems.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: tests/realtikvtest/addindextest/integration_test.go [hunk 1]
```diff
@@ -144,12 +144,12 @@ func TestIngestMVIndexOnPartitionTable(t *testing.T) {
 	tk.MustExec("use addindexlit;")
```
Comment (hawkingrei) [inline tests/realtikvtest/addindextest/integration_test.go:144]: It is the biggest problem of ```addindextest``` that it is too big for this test package to complete in a short time. the ```timeout``` in the bazel config is ```long```. two component' timeout config is ```long``` in the tidb repo. I suggest we can split tests into new package such as ```pitr```.

**Review by xiongjiwei — APPROVED** [review]

File: tests/realtikvtest/addindextest/integration_test.go (hunk 1 above)
Comment (tangenta) [reply tests/realtikvtest/addindextest/integration_test.go:144]: --- PASS: TestIngestMVIndexOnPartitionTable (452.41s)

The other tests in this package only take less than 1 min to finish.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**